### PR TITLE
Add runtime metrics support

### LINKED ISSUE #784: Prometheus View of runtime.MemStats
Motivation
-----------

I'm not sure if this belongs in this repo but I think there's some common runtime stats that most monitored services would want to expose similar to those available in `/debug/pprof`. I would see this feature being responsible for creating a measurement and view for each of the fields in the `runtime.MemStats` struct.

Impact
-------

As this would be a new function this is an API expansion. The performance impact of adding this view is expected to be approximately equivalent to the cost of calling `runtime.ReadMemStats` and populating all of the measurements.

Proposed Implementation
--------------------------

Add a package `go.opencensus.io/stats/view/runtime` with the following functions and data structs;

- `New(prefix string) RuntimeViews`.
- `RuntimeViews.Views() []View` - returns a collection of all of the runtime views.
- `RuntimeViews.Start(freq uint64)` - starts the background polling goroutine.
- `RuntimeViews.Stop()` - stops the background polling goroutine.

**param - prefix**

- Used to prefix each of the stats.

**param - freq**

- Not sure if this is desirable or if it should make the `ReadMemStats` call on each request.
- Idea would be to periodically call `ReadMemStats` in a background goroutine and cache the values until the timer has triggered.

**return - []View**

- Each element in the `[]View` would correspond to a field in the MemStats struct.
- Descriptions would be the first line of each of the fields in the MemStats struct documentation.
- The full names would be `${prefix}/runtime/${SNAKE_CASE_FIELD_NAME}` (I'll map these out more fully later).

### PR DESCRIPTION
**Pull request by panzerfahrer**
This adds a `metricproducer.Producer` implementation under `/plugin/` which enables the collection of runtime metrics.
 
Fixes #784

### CODE REVIEW

File: plugin/runmetrics/producer.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+	return c.reg.Read()
+}
+
+func newMemStats(producer *Producer) (*memStats, error) {
+	var err error
+	memStats := &memStats{}
+
+	// General
+	memStats.memAlloc, err = producer.createInt64GaugeEntry("mem_alloc", "Bytes of allocated heap objects", metricdata.UnitBytes)
```
[the remark this answers is not in the document]
Comment (panzerfahrer) [reply plugin/runmetrics/producer.go]: Could you please check if all names are ok with my last change.

**Comment by dprotaso** [comment]
Any thoughts on adding `GC` metrics?

ie. https://golang.org/pkg/runtime/debug/#GCStats

The Prometheus `GoCollector` exports it his way
https://github.com/prometheus/client_golang/blob/master/prometheus/go_collector.go#L319-L328

File: plugin/runmetrics/producer.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+	if options.EnableCPU {
+		producer.cpuStats, err = newCPUStats(producer)
+		if err != nil {
+			return err
+		}
+	}
+
+	Disable() // ensure no duplicate registrations
+	metricproducer.GlobalManager().AddProducer(producer)
+	return nil
```
Comment (rghetia) [inline plugin/runmetrics/producer.go]: There is a race condition here. Without any mutex there is a possibility of adding two producers.
Maybe you can create a singleton with sync.Once and with mutex delete/add producer when it is enabled/disabled. 
You can also create all entries once and simply based on boolean (memory/cpu) read those metrics.

File: plugin/runmetrics/producer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rghetia) [reply plugin/runmetrics/producer.go]: the names look fine. I didn't see cpu_seconds similar to one in [vmmetricreceiver](https://github.com/open-telemetry/opentelemetry-service/blob/559835ceeee85e5fa9ffcd5de025aa3b9c301f57/receiver/vmmetricsreceiver/vm_metrics_constants.go#L47)

**Pull request merged**